clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/lifecycle/LaunchFailureNotificationCleanupAgent.java

```diff
@@ -0,0 +1,179 @@
+                                        ServerGroupTagger serverGroupTagger) {
+    this.amazonClientProvider = amazonClientProvider;
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.serverGroupTagger = serverGroupTagger;
+  }
+
+  @Override
+  public String getAgentType() {
+    return LaunchFailureNotificationCleanupAgent.class.getSimpleName();
+  }
+
+  @Override
+  public String getProviderName() {
+    return AwsProvider.PROVIDER_NAME;
+  }
+
+  @Override
+  public long getPollIntervalMillis() {
+    return TimeUnit.MINUTES.toMillis(5);
+  }
+
+  @Override
+  public long getTimeoutMillis() {
+    return -1;
+  }
+
+  @Override
+  public void run() {
+    Collection<EntityTags> taggedEntities = serverGroupTagger.taggedEntities(
+      AmazonCloudProvider.ID,
+      null, // all accounts
+      TAG_NAME,
+      MAX_RESULTS
+    );
+
+    Map<String, AmazonAutoScaling> amazonAutoScalingClients = new HashMap<>();
+
+    taggedEntities.forEach(entityTags -> {
+      EntityTags.EntityRef entityRef = entityTags.getEntityRef();
+
+      AmazonAutoScaling amazonAutoScaling = getAmazonAutoScalingClient(
+        amazonAutoScalingClients, entityRef.getAccount(), entityRef.getRegion()
+      );
+
+      try {
+        if (hasLaunchFailures(amazonAutoScaling, entityTags)) {
+          return;
+        }
+
+        serverGroupTagger.delete(
+          AmazonCloudProvider.ID,
+          entityRef.getAccountId(),
+          entityRef.getRegion(),
+          entityRef.getEntityId(),
+          TAG_NAME
+        );
+      } catch (Exception e) {
+        log.error("Unable to determine if '{}' has launch failures", entityTags.getId(), e);
+      }
+    });
+  }
+
+  /**
+   * Fetch scaling activities and determine if the most recent activity was successful.
+   *
+   * A successful scaling activity is sufficient to indicate that a server group is no longer having launch failures.
+   */
+  protected boolean hasLaunchFailures(AmazonAutoScaling amazonAutoScaling, EntityTags entityTags) {
+    EntityTags.EntityRef entityRef = entityTags.getEntityRef();
+
+    try {
+      DescribeScalingActivitiesResult describeScalingActivitiesResult = amazonAutoScaling.describeScalingActivities(
+        new DescribeScalingActivitiesRequest().withAutoScalingGroupName(entityRef.getEntityId())
+      );
+
+      List<Activity> activities = describeScalingActivitiesResult.getActivities();
+      return !activities.isEmpty() && !activities.get(0).getStatusCode().equals(ScalingActivityStatusCode.Successful.toString());
+    } catch (Exception e) {
+      AmazonServiceException amazonServiceException = amazonServiceException(e);
+      if (amazonServiceException != null) {
+        if (amazonServiceException.getErrorMessage().toLowerCase().contains("name not found")) {
+          return false;
+        }
+      }
+
+      throw e;
+    }
+  }
+
+  private AmazonAutoScaling getAmazonAutoScalingClient(Map<String, AmazonAutoScaling> cachedClients,
```

Comment:
no need to cache the clients AmazonClientProvider already keeps them in a guava loadingcache